RxJavaTemplate/app/src/main/java/co/nimblehq/rxjava/ui/screens/webview/AppWebChromeClient.kt

```diff
@@ -0,0 +1,26 @@
+package co.nimblehq.rxjava.ui.screens.webview
+
+import android.webkit.WebChromeClient
+import android.webkit.WebView
+
+class AppWebChromeClient(private val onProgress: (WebViewProgress) -> Unit) : WebChromeClient() {
+
+    override fun onProgressChanged(view: WebView?, newProgress: Int) {
+        super.onProgressChanged(view, newProgress)
+        val progress = when (newProgress) {
+            PROGRESS_STARTED -> WebViewProgress.Show
+            PROGRESS_COMPLETED -> WebViewProgress.Hide
+            else -> WebViewProgress.Progress(newProgress)
+        }
+        onProgress.invoke(progress)
+    }
+}
+
+sealed class WebViewProgress {
+    object Show : WebViewProgress()
+    data class Progress(val progress: Int) : WebViewProgress()
```

Comment:
🤔  I think we could combine `Show` and `Progress` rather than separate 2 states. How about this one?

`Show(val progress: Int)`

Reply:
@hoangnguyen92dn I made a refactor to handle [this](https://github.com/nimblehq/android-templates/pull/66#discussion_r615677041) review from @minhnimble, get rid of [_showLoading usage as protected](https://github.com/nimblehq/android-templates/pull/66/commits/eb10164e129f40ab0a889a713a5c481c77822041#diff-e1c14270a4061009e91f46de7ed6a78c7dbda11be90ba748d043f7702ee9c6daR15)